hutch_python/base_plugin.py

```diff
@@ -18,15 +18,22 @@ class BasePlugin:
     priority = 0
     name = None
 
-    def __init__(self, conf):
+    def __init__(self, conf=None, info=None):
         """
+        Arguments will be passed every time by the loader, but are optional for
+        ease of testing and to help create metaplugins. At least one or the
+        other must be provided.
+
         Parameters
         ----------
-        conf: dict
+        conf: dict, optional
             The full dict from the yaml file.
+
+        info: object, optional
+            The portion of the dict specific to this plugin
         """
         self.conf = conf
-        self.info = conf[self.name]
+        self.info = info or conf[self.name]
```

Comment:
I think this should be
```python
info or conf.get(self.name, {})
```
The current setup would raise a KeyError if `info` is not passed and `Plugin.name` is not in `conf`. That means for instance you couldn't pass this directly to Questionnaire Plugin
```
experiment:
   run : 15
   proposal: X123
```

Reply:
You're right that this is a problem. I'll change it to `self.info = info or conf.get(self.name)` tomorrow.  The default `None` return is fine here because there is no guarantee as to the type of `info`, it can be `dict`, `list`, or `str` depending on the plugin.

Reply:
👍